ext/ffi/ffi.stub.php

```diff
@@ -10,15 +10,15 @@ public static function cdef(string $code = "", ?string $lib = null): FFI {}
 
     public static function load(string $filename): ?FFI {}
 
-    public static function scope(string $name): ?FFI {}
+    public static function scope(string $name): FFI {}
 
     public static function new(FFI\CType|string $type, bool $owned = true, bool $persistent = false): ?FFI\CData {}
 
     /** @prefer-ref $ptr */
     public static function free(FFI\CData $ptr): void {}
 
     /**
-     * @param FFI\CData|string|int|null $ptr
+     * @param FFI\CData|int|float|bool|null $ptr
```

Comment:
what's the problem with `string`?

Reply:
I think the relevant code is https://github.com/php/php-src/blob/57430e6f1eb58e71c64fd9f66bd13a916bdb8b23/ext/ffi/ffi.c#L3833-L3870

There are legs for `< IS_STRING`, `== IS_LONG` and `== IS_NULL`, so `== IS_STRING` would throw, wouldn't it?

Reply:
Ah, yes, you are right. I didn't notice that it's for the 2nd parameter, not the 1st one.

Reply:
The error message seems to be very weird though based on `zend_wrong_parameter_class_error(2, "FFI\\CData", zv);`...  I don't see any reason why `IS_STRING ` is excluded from the first condition, but that's what we have now.